pkg/policies/opa/rego/aws/aws_s3_bucket/noS3BucketSseRules.rego

```diff
@@ -1,9 +1,56 @@
 package accurics
 
-{{.prefix}}noS3BucketSseRules[retVal] {
-	bucket := input.aws_s3_bucket[_]
-    bucket.config.server_side_encryption_configuration == []
-    rc = "ewogICJzZXJ2ZXJfc2lkZV9lbmNyeXB0aW9uX2NvbmZpZ3VyYXRpb24iOiB7CiAgICAicnVsZSI6IHsKICAgICAgImFwcGx5X3NlcnZlcl9zaWRlX2VuY3J5cHRpb25fYnlfZGVmYXVsdCI6IHsKICAgICAgICAic3NlX2FsZ29yaXRobSI6ICJBRVMyNTYiCiAgICAgIH0KICAgIH0KICB9Cn0="
-    traverse = ""
-    retVal := { "Id": bucket.id, "ReplaceType": "add", "CodeType": "block", "Traverse": traverse, "Attribute": "server_side_encryption_configuration", "AttributeDataType": "base64", "Expected": rc, "Actual": null }
+{{.prefix}}noS3BucketSseRules[bucket.id] {
+    bucket := input.aws_s3_bucket[_]
+    checkSSEConfig(input, bucket)
 }
+
+checkSSEConfig(inobj, bucket) {
+    object.get(bucket.config, "server_side_encryption_configuration", "undefined") == [[], null, "undefined"][_]
+    object.get(inobj, "aws_s3_bucket_server_side_encryption_configuration", "undefined") == [[], null, "undefined"][_]
+}
+
+checkSSEConfig(inobj, bucket) {
+    object.get(bucket.config, "server_side_encryption_configuration", "undefined") == [[], null, "undefined"][_]
+    object.get(inobj, "aws_s3_bucket_server_side_encryption_configuration", "undefined") != [[], null, "undefined"][_]
+
+    sse := inobj.aws_s3_bucket_server_side_encryption_configuration[_]
+    matchSSE(bucket.id, sse.config)
+}
+
+matchSSE(bucketid, sseConfig) {
+    cleanID := cleanSSEBucketID(sseConfig.bucket)
+    bucketid != cleanID
+}
+
+matchSSE(bucketid, sseConfig) {
+    cleanID := cleanSSEBucketID(sseConfig.bucket)
+    bucketid == cleanID
+    checkSSE(sseConfig)
+}
+
+checkSSE(sseConfig) {
```

Comment:
this function is not requires. If resource is declared S3 bucket will have the default s3 (AWS) managed sse is enabled.